Runtime/BacktraceClient.cs

```diff
@@ -930,25 +930,34 @@ internal void HandleLowMemory()
         /// <param name="type">log type</param>
         internal void HandleUnityMessage(string message, string stackTrace, LogType type)
         {
-            if (!Enabled)
+            if (!Enabled || !Configuration.HandleUnhandledExceptions)
             {
                 return;
             }
-            var unityMessage = new BacktraceUnityMessage(message, stackTrace, type);
-            if (Configuration.HandleUnhandledExceptions && unityMessage.IsUnhandledException())
+            if (string.IsNullOrEmpty(message) || (type != LogType.Error && type != LogType.Exception))
             {
-                BacktraceUnhandledException exception = null;
-                var invokeSkipApi = true;
-
-                // detect sampling flow
-                // we should apply sampling only to unhandled exceptions that are type LogType == Error
-                // log type error won't provide full exception information
-                if (type == LogType.Error && SamplingShouldSkip())
+                return;
+            }
+            BacktraceUnhandledException exception = null;
+            var invokeSkipApi = true;
+            // detect sampling flow
+            // we should apply sampling only to unhandled exceptions that are type LogType == Error
```

Comment:
My understanding here is we only sample LogType.Error messages because they can be quite spammy. For other exception/error types we will always send and never do any sampling. And also we can use filtering to filter out certain kinds of reports.

Reply:
Thats true - I think we should adjust the comment to be sure that it points to LogType.Error not to unhandled exceptions, right?